crates/aptos-telemetry-service/src/auth.rs

```diff
@@ -52,15 +52,22 @@ pub async fn handle_auth(
     let (remote_public_key, handshake_state, _payload) = context
         .noise_config()
         .parse_client_init_message(&prologue, client_init_message)
-        .map_err(|_| reject::reject())?;
+        .map_err(|e| {
+            debug!("error performing noise handshake: {}", e);
+            reject::custom(ServiceError::invalid_request_body(
+                "error performing handshake",
+            ))
+        })?;
 
     let (epoch, peer_role) = match context.validator_cache().read().get(&body.chain_id) {
         Some((epoch, peer_set)) => {
             match peer_set.get(&body.peer_id) {
```

Comment:
Can it fail here in matching peer_id?  You may want to throw another error if that is the case.

Reply:
Also, how do we make sure the epoch is the current epoch?

Reply:
Yes, it can fail to get matching peer_id, and we handle it as well in the nested None block.

During authentication, we assume that the validator cache is up-to-date and if we can find the peer in the validator set, then we know its part of that epoch, and generate an auth token with the epoch.

After that, for each request (metrics ingest, big query ingest, etc), we compare the epoch in the auth token to the validator cache epoch and if they differ, we return a 401, and ask client to authenticate again.